cdc/kv/client.go

```diff
@@ -1343,16 +1353,16 @@ func (s *eventFeedSession) sendResolvedTs(
 				continue
 			}
 			slot := worker.inputCalcSlot(regionID)
-			statefulEvents[slot] = append(statefulEvents[slot], &regionStatefulEvent{
-				resolvedTs: resolvedTs,
-				regionID:   regionID,
-				state:      state,
-			})
+			statefulEvents[slot].resolvedTsEvent.regions = append(
+				statefulEvents[slot].resolvedTsEvent.regions, state,
+			)
+			// regionID is just an slot index
+			statefulEvents[slot].regionID = regionID
```

Comment:
Why not set the `regionID` first and then append it to the slice? Which can eliminate duplication slice index: `statefulEvents[slot]`.